### PR TITLE
NBA: adding TypeError checks around operations

### PR DESCRIPTION
**Pull request by SeanAmmirati**
This small PR simply adds some exception handling to some of the calculations for the NBA boxscore classes. 

I also intend to do this for the other classes as well. 

This was done because in the current state, running something like the following would raise a `TypeError`: 

```python
from sportsreference.nba.boxscore import Boxscore

bs = Boxscore('194910290TRI')
df = bs.dataframe
```

This was because the information that would have been required for all of these operations were valued None. While the decorators handle this issue for the particular values, the `TypeError` would happen with the arithmetic operators. 

This assumes that the attributes will only be of the correct type (float or int) or NoneType. 

This should also be done for the other classes. It appears that this will be a problem for very old games. For reference, this was the first NBA basketball game available on basketball reference. 

Happy to discuss this further or perhaps consider other ways to handle this. Also will be happy to do this for the other classes as I get to them.

### CODE REVIEW

File: sportsreference/nba/boxscore.py [hunk 1]
```diff
@@ -976,9 +986,14 @@ def away_two_point_field_goal_percentage(self):
...
-            float(self.away_two_point_field_goal_attempts)
-        return round(float(result), 3)
+        try:
+            result = float(self.away_two_point_field_goals) / \
+                float(self.away_two_point_field_goal_attempts)
+        except TypeError:
+            result = None
+        else:
+            result = round(float(result), 3)
+        return result
```
Comment (roclark) [inline sportsreference/nba/boxscore.py:996]: Beautiful! 😃

**Comment by SeanAmmirati** [comment]
Hi @roclark, thanks for the quick response! 

Happy to help in any way I can. Thanks for creating this package -- it is incredibly useful and helpful. 

More than happy to add the test cases, will do that shortly and check back once this has been done. Also apologies, I force pushed another small edit to avoid creating another commit, just fyi in case there are any conflicts in the history. 

Thanks!

**Comment by SeanAmmirati** [comment]
Just added some unit tests.

Please let me know if this formatting works or if you have any suggestions. 

Also, thanks for the note on the profile picture, GLaDOS! The cake is a lie! 👍 

Thanks! 

Sean

**Comment by roclark** [comment]
Hey @SeanAmmirati, sorry for not getting back to you earlier - it's been a very long week. These updates look great, and I appreciate the work on the test cases! Also, I'm happy to see the the codecov reports are now working on forked updates. This PR looks in great shape to me, so I will go ahead and merge it with master.

Thank you again both for the update and for your patience! This is absolutely wonderful!

**Pull request merged**